### ISSUE TITLE
Kotlin\Native IR plugin

### ISSUE THREAD

**Issue by ElisaTronetti** [bnorm/debuglog#6]
Hi!
I'm currently working on my master thesis and I have to create a compiler plugin. Since yours is kinda all the documentation that exists about kotlin compilers plugin, I am using as a reference your project.
I have to make my compiler plugin work with Kotlin\JS, Kotlin\JVM and Kotlin\Native. In your blog you wrote that it is necessary to write a different gradle plugin to make Kotlin\Native work, but in your debuglog project you don't use it. 
I wanted to ask if there is not the gradle plugin because native is not in your targets or because it is not necessary anymore. 
I'm asking this because currently in my project the compilations doesn't end successfully, because only Kotlin\Native throws a NullPointerException (specifically when I try to reference the annotation class).

I'm sorry to bother you, but since the documentation is nowhere to be found, I thought that this question might be also useful to other people.

**Comment by bnorm** [comment]
Hello! Not a bother at all, hopefully I can provide sufficient answers.

> In your blog you wrote that it is necessary to write a different gradle plugin to make Kotlin\Native work, but in your debuglog project you don't use it. I wanted to ask if there is not the gradle plugin because native is not in your targets or because it is not necessary anymore.

The blog is a little bit out of date. As of Kotlin 1.7, it is no longer required to have a separate Kotlin/Native artifact for your compiler plugin. Hence why it was deleted out of this sample plugin. This repository is setup to support all Kotlin platforms (JVM, JS, and Native) so should be a good example there (though I only build against 1 Kotlin/Native target: the current OS).

> I'm asking this because currently in my project the compilations doesn't end successfully, because only Kotlin\Native throws a NullPointerException (specifically when I try to reference the annotation class).

Maybe you are using a Kotlin version <1.7 and still need the native artifact? Without more details on how it is failing, I'm not sure I can be much help. If you can link a repository I could take a quick look and see if I notice anything strange.

**Comment by ElisaTronetti** [comment]
Hello again!
Since I wasn't so sure about Kotlin\Native yesterday, I tried to exclude it from my targets. It turned out that I had the same problem with all of my targets. The problem that caused my NullPointerException was the wrong reference to the annotation class.
After your answer I tried to put back Kotlin\Native as a target, which wasn't the problem after all. Thankfully it also works!

Thank you so much for your work and your answer, it was actually pretty helpful!
